tools/NetCoreConversion/NetCoreConversion/Utility/ListExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NetCoreConversion.Utility
+{
+    internal static class ListExtensions
+    {
+        public static IEnumerable<string> UpdateFirstAndLast(this List<string> list, string firstLeader, string lastFollower) => 
+            list.Select((value, i) => $"{(i == 0 ? firstLeader : String.Empty)}{value}{(i == list.Count - 1 ? lastFollower : String.Empty)}");
```

Comment:
Not a required change

Reply:
Many algorithms involve only changing the first element and the last element in a series of elements. Unfortunately, C# doesn't lend itself to doing this easily. Also, you can't use that `Concat` logic since that is adding an element to the start and end of `list` instead of conditionally adding information to the start and end of every value in `list`.